addons/binding/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/internal/dpt/KNXCoreTypeMapper.java

```diff
@@ -534,6 +551,25 @@ public KNXCoreTypeMapper() {
         defaultDptMap.put(DateTimeType.class, DPTXlatorTime.DPT_TIMEOFDAY.getID());
         defaultDptMap.put(StringType.class, DPTXlatorString.DPT_STRING_8859_1.getID());
         defaultDptMap.put(HSBType.class, DPTXlatorRGB.DPT_RGB.getID());
+
+        defaulItemTypeMap.put(DateTimeType.class, DateTimeItem.class);
```

Comment:
You are effectively duplicating (partially) the information from acceptedItemTypes/acceptedCommandTypes here. Not sure if that's a good idea...

Reply:
In the advanced binding I do implement I use a ChannelBuilder to create Channels based on the information stored in the ETS5 files. ChannelBuilder::create() takes an acceptedItemType as input, which is provided by the defaulItemTypeMap you mention, and for which in turn the Key is provided by KNXCoreTypeMapper::toTypeClass() based on the DPT that is derived from the ETS5 data.